### PR TITLE
#3265 Add CLIMEDO functionalities to cases

### LINKED ISSUE #3265: [CLIMEDO Integration] Add the CLIMEDO Functionality to cases
### Situation Description
Currently the symptom diary can only be used for contacts. With this story we extend the functionality to cases

### Feature Description
Add the functionality of creating users, sync personal data and endfollowupdate, and receive data from Symptom Journal for cases and the follow-up for cases

### PR DESCRIPTION
**Pull request by vlad-ciucescu**
closes #3265 
**NOTE:** uses commit from branch `feature-3294-Inform-External_Journal-about-canceled-or-completed-follow-ups`. This should probably be merged after #3294 is merged.

### CODE REVIEW

File: sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonFacadeEjb.java [hunk 1]
```diff
@@ -392,8 +399,82 @@ public Date getLatestFollowUpEndDateByUuid(String uuid) {
 		cq.orderBy(cb.desc(contactRoot.get(Contact.FOLLOW_UP_UNTIL)));
 
 		List<PersonFollowUpEndDto> resultlist = em.createQuery(cq).getResultList().stream().distinct().collect(Collectors.toList());
-		return resultlist.get(0).getLatestFollowUpEndDate();
```
Comment (FredrikSchaeferVitagroup) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonFacadeEjb.java:395]: This method needs to also consider FollowUpEndDates of cases here, if i'm not mistaken...

**Review by FredrikSchaeferVitagroup — DISMISSED** [review]

File: sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonFacadeEjb.java (hunk 1 above)
Comment (vlad-ciucescu) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonFacadeEjb.java:395]: So should the latest follow up end date from both contacts and cases related to the person be returned?
What about the method above, `getLatestFollowUpEndDates`? Should we return a list with dates combined from both cases and contact?

Comment (FredrikSchaeferVitagroup) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonFacadeEjb.java:395]: Yes, good point. Each of those methods should return the latest follow up end date for persons, no matter if they come from a contact or a case.

**Comment by FredrikSchaeferVitagroup** [comment]
If we add cases to external journals, this should be reflected in unit tests, for example because they change logic for latestFollowUpEndDates of persons. Relating tests are implemented in #3195.

**Comment by vlad-ciucescu** [comment]
> 
> 
> If we add cases to external journals, this should be reflected in unit tests, for example because they change logic for latestFollowUpEndDates of persons. Relating tests are implemented in #3195.

I updated the tests in `PersonFacadeEjbTests` which were related to the followup date. The tests in `ExternalJournalServiceTest` do not contain any reference to contacts.

**Comment by FredrikSchaeferVitagroup** [comment]
The code looks good to me now. Please resolve merge conflicts.

**Comment by FredrikSchaeferVitagroup** [comment]
I talked to the climedo developers about it, they said this issue needs further refinement.

**Comment by FredrikSchaeferVitagroup** [comment]
In a diskussion with climedo it came out that they changed their app. It seems that it happens to just fit to what we implemented here (they removed the different study for cases, all cases and contacts use the same study now).
Please update the branch to the latest state of development!

**Review by vidi42 — APPROVED** [review]

**Pull request merged**